futures-util/src/async_await/random.rs

```diff
@@ -39,16 +36,75 @@ fn random() -> u64 {
             hasher.write_usize(COUNTER.fetch_add(1, Ordering::Relaxed));
             seed = hasher.finish();
         }
-        seed
+        seed as usize
```

Comment:
So, it turns out that `DefaultHasher` doesn't actually involve any randomness, so you could replace this with `13646096770106105413` (for the current algorithm backing it).

Alternatively, this could be changed to use `RandomState` instead of `DefaultHasher` so that there is actually some randomness injected. That should probably be a separate PR though.

Reply:
Hm, yeah without randomness there's presumably little point in using a hasher at all, the seed generator may as well just be reduced to `COUNTER.fetch_add(1)` and give out sequential seeds at that point.